### DISCUSSION TITLE
Character used for a single bond is the same as that for a negative charge, substitute negative charges with 'minus' instead of 'minus-hyphen'

### DISCUSSION THREAD

**Discussion by pv-is-nrt** [rdkit/rdkit#5726] [Q&A]
I see that the character used for a single bond is the same as that for a negative charge. This is presenting problems while parsing a smiles string correctly that contains both. Is there a reason why the characters cannot be different?

Can someone please help me with a solution where I can define my own different character for a negative charge while using the ```Chem.MolToSmiles(mol, isomericSmiles=True, allBondsExplicit=True)``` method?

Edit: Adding an example of what I need.
```python
from rdkit import Chem

# here's a smiles string from ChEMBL database
# note that this has both negative charges and single bonds. Single bonds do not always come next to an opening parentheses and -ve charges, I think, will always be within [ ]. But what if a single bond is present inside [ ] too? 
smiles = 'COC(=O)N[C@H](C(=O)N1C[C@H](N=[N+]=[N-])C[C@H]1c1ncc(-c2ccc(-c3ccc(-c4[nH]c([C@@H]5C[C@@H](N=[N+]=[N-])CN5C(=O)[C@@H](NC(=O)OC)C(C)C)nc4Cl)cc3)cc2)[nH]1)C(C)C'

# convert the smiles to a mol
# I understand that it is here that RDKit is capable at parsing the smiles and it knows how to differentiate the negative charge from the bond. 
mol = Chem.MolFromSmiles(smiles)

# convert the mol back to smiles with bonds present
# I need the bonds explicitly shown for my use case
# Again, perhaps it is here that I can modify the method and use my own symbol for a negative charge?
expanded_smiles = Chem.MolToSmiles(mol, isomericSmiles=True, allBondsExplicit=True)
expanded_smiles
# outputs
# C-O-C(=O)-N-[C@H](-C(=O)-N1-C-[C@H](-N=[N+]=[N-])-C-[C@H]-1-c1:n:c:c(-c2:c:c:c(-c3:c:c:c(-c4:[nH]:c(-[C@@H]5-C-[C@@H](-N=[N+]=[N-])-C-N-5-C(=O)-[C@@H](-N-C(=O)-O-C)-C(-C)-C):n:c:4-Cl):c:c:3):c:c:2):[nH]:1)-C(-C)-C
```

**Comment by bp-kelley** [comment]
This seems like a bad idea.  Smiles is a fully defined grammar where the symbols are context dependent, this includes numbers as well.  A charge symbol will only occur in brackets for instance, numbers can occur for hydrogen count, charge, atom maps and ring closures to name a few.

When you say parsing, what are you actually trying to do, obviously, the RDKit can parse smiles.

**Reply by pv-is-nrt** [reply]
Thanks for replying and sure, but why use the same symbol?
I added an example to show what I am trying to do.

**Reply by DavidACosgrove** [reply]
Perhaps https://pubs.acs.org/doi/pdf/10.1021/ci00057a005# will explain some of the motivation for the grammar and symbol choices. Personally, I’ve been using SMILES for 30-odd years and never found the two uses of - confusing.

**Reply by pv-is-nrt** [reply]
Oh I understand and thanks for sharing that paper. The grammar is not confusing to me, as a human. The parsing logic is also probably perfectly built into the RDKit library. I just don't know enough technically, or maybe I am blanking, to be able to accomplish what I asked; and that's what I am hoping to find help with. But it's useful to know that long-term users of smiles never needed to have those symbols different.
